### PR TITLE
UCS/STRING_BUFFER: Implement string buffer on top of ucs_array_t

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucs/datastruct/string_buffer.c [hunk 1]
```diff
@@ -17,116 +17,89 @@
...
-    ret       = vsnprintf(strb->buffer + strb->length, max_print, fmt, ap);
+    max_print = ucs_array_end_capacity(&strb->str);
+    ret       = vsnprintf(ucs_array_end(&strb->str), max_print, fmt, ap);
     va_end(ap);
 
     /* if failed, grow the buffer accommodate for the expected extra length */
     if (ret >= max_print) {
-        status = ucs_string_buffer_grow(strb, strb->length + ret + 1);
+        status = ucs_array_reserve(string_buffer, &strb->str,
+                                   ucs_array_length(&strb->str) + ret + 1);
```
Comment (hoopoepg) [inline src/ucs/datastruct/string_buffer.c:62]: minor: after this operation actual string buffer size will be ```size(original_array) + UCS_STRING_BUFFER_RESERVE + ret``` - we will have UCS_STRING_BUFFER_RESERVE non-used bytes

Comment (yosefe) [reply src/ucs/datastruct/string_buffer.c:62]: why? we didn't change the length during the 1st ucs_array_reserve(), it only increased the capacity
the 2nd time ucs_array_reserve() is used, ucs_array_length() will still be the original length

File: test/gtest/ucs/test_string.cc [hunk 2]
```diff
@@ -76,6 +76,18 @@ UCS_TEST_F(test_string_buffer, rtrim) {
     ucs_string_buffer_cleanup(&strb);
 }
 
+UCS_TEST_F(test_string_buffer, fixed) {
+    char buf[17];
+    UCS_STRING_BUFFER_FIXED(strb, buf);
+
+    ucs_string_buffer_appendf(&strb, "%s", "im");
+    ucs_string_buffer_appendf(&strb, "%s", "mrmeeseeks");
+    ucs_string_buffer_appendf(&strb, "%s", "lookatme");
```
Comment (dmitrygx) [inline test/gtest/ucs/test_string.cc:85]: 👍

**Pull request merged**